experimental/benchmark/src/main.rs

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+mod proto {
+    tonic::include_proto!("oak.examples.trusted_database");
+}
+
+mod backend;
+mod database;
+mod runtime;
+
+use crate::{backend::TestBackend, database::Database};
+use log::info;
+use runtime::TestRuntime;
+use std::time::Instant;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Clone)]
+#[structopt(about = "Trusted Database Client")]
+pub struct Opt {
+    #[structopt(long, help = "Database size (number of items)")]
+    database_size: usize,
+}
+
+#[tokio::main(core_threads = 4)]
+async fn main() -> anyhow::Result<()> {
+    // Set current dir to create `oak_tests` in the `experimental/benchmark` directory.
+    std::env::set_current_dir("./experimental/benchmark")?;
+
+    env_logger::init();
+    let opt = Opt::from_args();
+
+    let database = Database::new(opt.database_size);
+    let serialized_database = database.serialize();
+    let mut backend = TestBackend::start(&database, 8888_u16).await;
+
+    // Start Oak runtime.
+    info!(
+        "Loading Runtime, (sending database - size: {} bytes)",
+        serialized_database.as_bytes().len()
+    );
+    let runtime_loading_start = Instant::now();
+    let mut runtime = TestRuntime::start(&serialized_database).await;
+    let runtime_loading_time = (runtime_loading_start.elapsed().as_millis() as f64) / 1000.0;
+    info!("Runtime loaded: {:.3}s", runtime_loading_time);
+
+    // Measure raw request durations.
+    let iterations: u16 = 10;
+    let mut total_raw_request_time = 0;
+    for _ in 0..iterations {
+        let request_start = Instant::now();
+        let _ = backend.send_request("0").await?;
+        total_raw_request_time += request_start.elapsed().as_millis();
+    }
+    let raw_average_request_time = (total_raw_request_time as f64) / (iterations as f64) / 1000.0;
```

Comment:
Could not you just measure the time the entire iteration takes, instead of measuring the time for each request and then summing them?